### ISSUE TITLE
investigate possibility to start browsermob proxy for mobile before app startup

### ISSUE THREAD

**Issue by vdelendik** [zebrunner/carina#491]
we can't start proxy until we receive device from STF/HUB. But as result app usually can't be started successfully and need to be restarted using adb stop/launch activity. Such use-case is pretty complicated.
We should determine automatic solution for getting device from hub even before starting driver. It seems like that's the only way to get capabilities with proxy_port preliminary.

Moreover such feature should help in resolution for:
https://github.com/qaprosoft/carina/issues/445
https://github.com/qaprosoft/carina/issues/436
https://github.com/qaprosoft/carina/issues/382

**Comment by vdelendik** [comment]
won't fix. the only change we can do: start proxy preliminary if we already know browsermob_port somehow. If not then we have to find device based on capabilities, install and launch app, get proxy_port from slotCapabilities and restartDriver(true) on the same to reuse proxy during app restart

**Comment by vdelendik** [comment]
we started investigation howto improve carina-grid (QPS-HUB) to make integration between carina and devices smarter: https://github.com/qaprosoft/carina/issues/184.
As only #184 is finished we can start with current one

**Comment by vdelendik** [comment]
closing as won't fix because browermob and browserup deprecated and will be removed completely from carina. Scalable mitm proxy should be added on Zebrunner Selenium Grid side (https://zebrunner.com/selenium-grid) and maybe device farm (https://zebrunner.com/mcloud)
